designdocs/CAPACITY.md

```diff
@@ -0,0 +1,148 @@
+   type Balance: Balance;
+   
+	 fn replenish_by_account(msa_id: MessageSenderId, amount: Balance) -> Result<Balance, Eroor> {};
+   fn replenish_all_for_account(msa_id: MessageSenderId) -> Result<Balance, Error>;
+   fn can_replenish(msa_id: MessageSenderId) -> bool;
+}
+```
+
+**How capacity replenishes?**
+
+As mentioned above, epoch periods are composed of blocks. The next epoch period calculated is based on the current epoch fullness, and expands and contracts depending on a threshold. This threshold is configurable and can be called `config::epochThreshold` . Additionally, we can also configure a multiplier that we increase or decrease the next epoch based on the fullness of the current epoch. This is defined as `config::epochUtilizationMultiplier`.
+
+![Untitled](https://user-images.githubusercontent.com/3433442/171747512-651112be-bbdc-4197-96ef-0cd208f702db.png)
+
+The above illustrates two epochs where the second one contracts because network congestion has decreased. As a result of epoch decreasing, capacity is replenish faster. Note that initially epoch periods will be fixed in sized for a period of time.
+
+**How are epochs calculated?**
+
+Epoch are used to manage congestion on the network. As demand increases for network resources the epoch length increases which results in capacity taking longer to replenish. 
+
+Upon the finalization of each block, we can get the total amount of weight used and update the total amount of weight for an epoch
+
+![Untitled](https://user-images.githubusercontent.com/3433442/171747526-566ee44a-194f-47e3-8abb-e2c415ac7fb5.png)
+
+**Transactions fees**
+
+When submitting capacity transactions, before including these transaction in a block they are validated at the transaction pool. The validation implemented doing a [SignedExtension](https://docs.rs/sp-runtime/latest/sp_runtime/traits/trait.SignedExtension.html) that validates that the signed transaction is associated to an MSA that contains a Capacity balance. If the MSA has capacity it then increases the amount of capacity is used. If no Capacity is remaining for the epoch period the transaction remains in the future queue until the next epoch. Once the next epoch starts and Capacity becomes available the transaction is processed.
+
+Non-capacity transaction remained unchanged and follow the same default flow. During implementation a wrapper capacity transaction pallet is used to wrap pallet transaction payment to toggle between capacity and non-capacity transactions and set the [validity of the transaction.](https://paritytech.github.io/substrate/master/sp_runtime/transaction_validity/struct.ValidTransaction.html)
+
+![Untitled](https://user-images.githubusercontent.com/3433442/171749900-21470787-b74f-44fa-b32d-ac06138f7616.png)
+
+
+## Non-Goals
+Staking details are left for another design document.
+
+## Benefits and Risk
+The benefits of implementing Capacity is that it allows applications to increase their users my reducing cost.
```

Comment:
Needs also benefits and risks of this particular design.

Reply:
We need to Know risks or potential risks we will  either mitigate at implementation time or expect to mitigate at some point.
Or are we saying there arent any risks to capacity model